### DISCUSSION TITLE
Any examples of using an array of objects?

### DISCUSSION THREAD

**Discussion by mhddngs** [47ng/nuqs#682] [Q&A] [answered]
Is this valid use? :

```ts
const ItemSchema = z.object({
  id: z.string(),
  groupId: z.string(),
});

export function usePurchasedItems() {
  const [purchasedItems, setPurchasedItems] = useQueryState(
    "item",
    parseAsArrayOf(parseAsJson(ItemSchema.parse))
  );
  return [purchasedItems, setPurchasedItems] as const;
}
```
typesense seems to recognize it as valid but I want to make sure. I'm getting nulls when trying to read from the array.

**Comment by franky47** [comment, marked as answer]
> I'm getting nulls when trying to read from the array.

The `purchasedItems` type here is `z.infer<typeof ItemSchema> | null`, because there can be no `item` key in the URL. To prevent that, set a default value:
```ts
const [purchasedItems, setPurchasedItems] = useQueryState(
  "item",
  parseAsArrayOf(parseAsJson(ItemSchema.parse)).withDefault([])
);
```

Now be aware that URLs are user input: I don't know what your exact use-case is, but know that users can (and will) modify URLs, so in your case, they could add, remove, or edit purchased items.

**Reply by mhddngs** [reply]
Thank you this is helpful. They are purchases in name only without any financial bearing :)
